### DISCUSSION TITLE
Technical Criteria

### DISCUSSION THREAD

**Discussion by luziusmeisser** [Frankencoin-ZCHF/Frankencoin#11] [Acceptable Collaterals]
The collateral must fulfill the following criteria:

- The token must adhere to the ERC-20 standard.
- The collateral must be freely transferrable. Mandatory allowlisting (also known as whitelisting) is not acceptable as that would jeopardize the challenge mechanism.
- The token's transfer function must revert on failure as required by the ERC-20 standard. The Frankencoin system ignores the return value of the transfer method. Tokens that returning "false" to indicate a failure cannot be accepted.
- There should be no admin functions in the token. Tokens that are upgradable or tokens that contain functions like "pause" or "freeze" can only be accepted if the holder of the admin keys is known and enjoys an outstanding reputation with regards to respecting the token holders rights. Furthermore, such tokens will be subject to a risk premium when calcualting interest rates and limits.
- ERC-777 tokens are not allowed. Frankencoin is not safe to use with collateral that supports transfer hooks.
- TODO: add further criteria :)

**Comment by samclassix** [comment]
- ERC-20 collateral decimals >= 24
- remove zchf and fps as collateral
> MintingHub:openPosition (hard code)
> if (collateralAddress == address(zchf)) --> fail
> if (collateralAddress == address(zchf.equity())) --> fail

**Comment by samclassix** [comment]
# New Position from MintingHub V2

When opening a new position, it is normal to get a warning. This warning is due to an internal check to ensure that the collateral token fails with an error on invalid transfers. To test whether it does so, the minting hub actually tries to perform an invalid transfer and then ignores the (expected) error. The transaction still goes through, but metamask and etherscan might show some warnings as they detect the ignored test error happening within the transaction.

![Screenshot 2024-12-10 at 9 58 02 AM](https://github.com/user-attachments/assets/532afb70-638a-4918-babc-8a596a2e1fa0)

> MintingHubV2:OpenPosition (Solidity)

# Wallet e.g. MetaMask Warning

<img width="342" alt="Screenshot 2024-12-10 at 9 56 58 AM" src="https://github.com/user-attachments/assets/d0b13d09-6588-4219-9223-fbd09e2f6db3">

# Explorer warning after successful Tx

<img width="884" alt="Screenshot 2024-12-10 at 9 55 32 AM" src="https://github.com/user-attachments/assets/02706eeb-a519-487e-aea3-3eba448a4f2f">
